stock_cycle_count/models/stock_cycle_count_rule.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-18 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from datetime import timedelta, datetime
+
+
+class StockCycleCountRule(models.Model):
+    _name = 'stock.cycle.count.rule'
+    _description = "Stock Cycle Counts Rules"
+
+    @api.multi
+    def _compute_currency_id(self):
+        for rec in self:
+            rec.currency_id = self.env.user.company_id.currency_id
+
+    @api.model
+    def _selection_rule_types(self):
+        return [
+            ('periodic', _('Periodic')),
+            ('turnover', _('Value Turnover')),
+            ('accuracy', _('Minimum Accuracy')),
+            ('zero', _('Zero Confirmation'))]
+
+    @api.multi
+    @api.constrains('rule_type', 'warehouse_ids')
+    def _check_zero_rule(self):
+        for rec in self:
+            if rec.rule_type == 'zero' and len(rec.warehouse_ids) > 1:
+                raise ValidationError(
+                    _('Zero confirmation rules can only have one warehouse '
+                      'assigned.')
+                )
+            if rec.rule_type == 'zero':
+                zero_rule = self.search([
+                    ('rule_type', '=', 'zero'),
+                    ('warehouse_ids', '=', rec.warehouse_ids.id)])
+                if len(zero_rule) > 1:
+                    raise ValidationError(
+                        _('You can only have one zero confirmation rule per '
+                          'warehouse.')
+                    )
+
+    @api.onchange('rule_type')
+    def _compute_rule_description(self):
```

Comment:
Please follow OCA convention for onchange function name

Reply:
it's both a compute method and an onchange method. https://github.com/Eficent/stock-logistics-warehouse/blob/b9b4c2283fdffebbcdb508e49e07a5e512d70cb6/stock_cycle_count/models/stock_cycle_count_rule.py#L100

Reply:
If you put
```python
@api.depends('rule_type')
```
The change on rule_type value will trigger the change on rule_description on the interface.

Reply:
done